devices/src/virtio/block.rs

```diff
@@ -166,20 +166,46 @@ impl Request {
             return Err(Error::UnexpectedWriteOnlyDescriptor);
         }
 
-        let request_type = request_type(&mem, avail_desc.addr)?;
-        let sector = sector(&mem, avail_desc.addr)?;
+        let mut req = Request {
```

Comment:
nit: I like the building the Object at the end as part of the 'return' non-mutably because it better enforces fully populating the object.

Building a mutable dummy `Request` and then incrementally overwriting the dummy fields with valid ones is more error-prone to forgetting to update something and ending up with dummy data.

Reply:
In theory, I would agree. However, in this particular instance, I think it makes the code more readable.

Reply:
In the previous version of the PR this was needed, in this one it's not and doesn't seem to improve readability either...
...but since it's just style preference I will concede to your preference.